manifest/docker_schema2_list.go

```diff
@@ -86,6 +86,30 @@ func (list *Schema2List) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (list *Schema2List) UpdateInstancesAndDiscardOthers(listUpdate []ListUpdate) error {
+	existingManifest := map[digest.Digest]Schema2ManifestDescriptor{}
+	for _, m := range list.Manifests {
+		for _, u := range listUpdate {
+			if m.Digest == u.Digest {
+				existingManifest[u.Digest] = m
+			}
+		}
```

Comment:
AFAICS this could unconditionally build `existingManifest[m.Digest] = m`; the entries to be removed would just be ignored. That turns this from O(N^2) into O(N).

Same in the other implementation.